src/shared/logs/logs.py

```diff
@@ -149,50 +152,106 @@ def get_raw_events(
-                    )
-                    .order_by("pgh_created_at")
-                    .values("pgh_created_at")[:1]
+        package_edit_qs = self._annotate_username(
+            PackageEditEvent.objects.filter(suggestion__in=suggestion_ids)
+        )
+
+        for package_edit_event in package_edit_qs.all().iterator():
+            raw_events.append(
+                PackageChangeEvent(
+                    suggestion_id=package_edit_event.suggestion.id,
+                    timestamp=package_edit_event.pgh_created_at,
+                    username=package_edit_event.username,
+                    action=package_edit_event.pgh_label,
+                    package_attribute=package_edit_event.package_attribute,
                 )
             )
-            .filter(proposal_id__in=suggestion_ids)
-        ).annotate(
-            package_names=ArrayAgg(
-                JSONObject(
-                    name="derivation__name",
-                    attribute=Replace(
-                        "derivation__attribute",  # type: ignore
-                        Concat(Value("."), "derivation__system"),  # type: ignore
-                        Value(""),
-                    ),
-                ),
-                distinct=True,
-            ),
-            package_count=Count("derivation__name", distinct=True),
+
+        maintainer_qs = self._annotate_username(
+            MaintainersEditEvent.objects.prefetch_related(
+                "pgh_context", "maintainer"
+            ).filter(suggestion__in=suggestion_ids)
         )
 
-        for package_event in package_qs.all().iterator():
+        for maintainer_event in maintainer_qs.all().iterator():
             raw_events.append(
-                PackageChangeEvent(
-                    suggestion_id=package_event.proposal_id,
-                    timestamp=package_event.pgh_created_at,
-                    username=package_event.username,
-                    action=package_event.pgh_label,
-                    package_names=package_event.package_names,
-                    package_count=package_event.package_count,
+                MaintainerChangeEvent(
+                    suggestion_id=maintainer_event.suggestion.id,
+                    timestamp=maintainer_event.pgh_created_at,
+                    username=maintainer_event.username,
+                    action=maintainer_event.pgh_label,
+                    # TODO: we should use Pydantic model for maintainers and
+                    # friends as well, at some point
+                    maintainer=cast(
+                        Maintainer, model_to_dict(maintainer_event.maintainer)
+                    ),
                 )
             )
 
         return sorted(raw_events, key=lambda event: event.timestamp)
 
+    def _remove_canceling_events(
+        self,
+        events: list[
+            PackageChangeEvent | SuggestionChangeEvent | MaintainerChangeEvent
+        ],
+        time_threshold_seconds: int = 300,
```

Comment:
```suggestion
        time_threshold_seconds: int = 30,
```

5 min is probably too wide of a window.